spec/requests/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe API::V1::UsersController, type: :request do
+  context 'when a user register' do
+    it 'returns the user' do
+      params = create_user_params
+      params[:email] = 'new_email@gmail.com'
+      post 'create', params: params
+
+      expect(JSON.parse(response.body)['data'].map { |item| item['type'] }).to contain_exactly('user', 'token')
+    end
+  end
+
+  context 'when a user register with an existing email' do
+    it 'receive an error' do
+      params = create_user_params
+      post 'create', params: params
```

Comment:
```suggestion

      post 'create', params: params
```

Reply:
Thanks! I did some experimentations to better understand "when", it's more clear now :)